tests/tls/test_self_signed_scheme.py

```diff
@@ -0,0 +1,154 @@
+import nutkit.protocol as types
+from tests.shared import (
+    get_driver_name,
+    TestkitTestCase,
+)
+from tests.tls.shared import (
+    TlsServer,
+    try_connect,
+)
+
+
+class TestSelfSignedScheme(TestkitTestCase):
+    """
+    Tests URL scheme neo4j+ssc/bolt+ssc where server is assumed to present a
+    signed server certificate but not necessarily signed by an authority
+    recognized by the driver.
+    """
+    def setUp(self):
+        super().setUp()
+        self._server = None
+        self._driver = get_driver_name()
+
+    def tearDown(self):
+        if self._server:
+            # If test raised an exception this will make sure that the stub
+            # server is killed and it's output is dumped for analysis.
+            self._server.reset()
+            self._server = None
+        super().tearDown()
+
+    schemes = "neo4j+ssc", "bolt+ssc"
+    feature_requirement = types.Feature.API_SSL_SCHEMES,
+    extra_driver_configs = {},
+
+    def _try_connect(self, scheme, host, driver_config):
+        return try_connect(self._backend, self._server, scheme, host,
+                           **driver_config)
+
+    def test_trusted_ca_correct_hostname(self):
+        """
+        A server certificate signed by a trusted CA should be accepted even when
+        configured for self signed.
+        """
+        self.skip_if_missing_driver_features(*self.feature_requirement)
+        for driver_config in self.extra_driver_configs:
+            for scheme in self.schemes:
+                with self.subTest(scheme
+                                  + "-" + str(driver_config)):
+                    self._server = TlsServer("trustedRoot_thehost")
+                    self.assertTrue(self._try_connect(scheme, "thehost",
+                                                      driver_config))
+                self._server.reset()
+
+    def test_trusted_ca_expired_server_correct_hostname(self):
+        """
+        A server certificate signed by a trusted CA but the certificate has
+        expired. Go driver happily connects when InsecureSkipVerify is enabled,
+        same for all drivers ?
```

Comment:
Should this comment be updated or is it left as a question intentionally?

Reply:
I just copy-pasted this. Not sure what the current state of the drivers is :thinking: